src/gui/widgets/subtitlelistwidget.cpp

```diff
@@ -470,11 +478,22 @@ void SubtitleListWidget::handleSecondaryTrackChange(int64_t sid)
     m_secondaryLock.unlock();
 }
 
-void SubtitleListWidget::addPrimarySubtitle(const QString &subtitle,
+void SubtitleListWidget::updatePrimarySubtitle(const QString &subtitle,
                                             const double   start,
                                             const double   end,
                                             const double   delay)
 {
+    // if parsing has been completed
+    // assume entry is in list and only look up
+    if (m_didParsePrimarySubs)
+    {
+        double timePos = GlobalMediator::getGlobalMediator()->getPlayerAdapter()->getTimePos();
+        auto target = --m_seenPrimarySubs.upperBound(timePos + TIME_DELTA);
```

Comment:
I must have made bad assumptions about how context works. My thinking process is that since I didn't touch any code relating to what happens when the user selects (multiple, which is what I understand context applies to) entries from the subtitle list, I didn't break things. Will look carefully at it